tools/azsdk-cli/docs/specs/99-repo-tooling-contract.md

```diff
@@ -0,0 +1,322 @@
+to AI agents.
+It seems likely that existing repository scripts will always exist in some capacity (see below), so a formalized approach to referencing
+these scripts will have the following benefits:
+
+- Improved velocity - tactical script changes can be made with confidence while not requiring an `azsdk` release.
+  - As `azsdk` matures and gains usage/impact, the validations/requirements to release a new version will increase.
+  - Small, language-specific changes can be made without needing to understand `azsdk` development.
+  - Breakglass changes can be made to resolve urgent issues (unblocking releases, security issues, etc.)
+- Easier migration of scripts to `azsdk`.
+  - It will take time to centralize automation into `azsdk`. There needs to be a bridge between `azsdk` and existing scripts that can support a gradual migration while maintaining reliability.
+- New automation experiments can be made in language repositories, even if the desired end state is a generic interface in `azsdk`.
+- Language specific edge cases can exist without cluttering the cli/mcp tool list.
+- Depending on calling a script (i.e. Export-Apis.ps1) that have a set of parameters unique to a repo will be fragile and make versioning `azsdk` with the source changes in the repo very difficult. It will also make this tool very difficult to use in other repo contexts (i.e. the openai or mcp repos).
+
+---
+
+## Goals and Exceptions/Limitations
+
+### Goals
+
+- [ ] Simplify SDK automation
+  - Make automation entrypoints easier to understand and invoke
+  - Invoke operations in the same way locally and in pipelines. Remove multiple entrypoints for the same logic.
+  - Reduce layers of indirection and complexity
+  - Improve testability of changes
+- [ ] Reduce tight coupling between `azsdk` and language repository automation
+  - `azsdk` cli commands and mcp tools should not be bound to a repository structure that isn't designed to version with it
+- [ ] Enable development in both areas
+  - Keep toil low for small, tactical changes to scripts in a repository
+  - Support an efficient on-ramp for script logic to be migrated to `azsdk` without requiring intensive release coordination
+
+### Limitations
+
+Managing parameter mappings can still cause tight coupling or potential breakages.
+The design proposal below outlines the use of `[Alias]` for powershell cmdlets to address this.
+However, this approach means that direct calls to repository tools (e.g. `msbuild`) should be made from the CLI, or in
+some special cases wrapped in a powershell script.
+
+Adding schemas for parameters could improve reliability but at the cost of requiring changes to `azsdk` and/or scripts to be coordinated.
+It would also require additional up front work to capture parameter schemas for existing scripts, along with CI checks to ensure the
+scripts implement the schema.
+
+## Design Proposal
```

Comment:
Just making sure that I'm following the end-to-end scenario being proposed.   The way that I'm reading this:

- There is one file in the language repositories related to directing DevEx platform commands

- All DevDex CLI and MCP tools will use this file to determine how to invoke repository-specific scripts

- The goal is to provide a neutral format for redirection from specific DevEx command to existing scripts.  For example, the `platform-update` command for .NET could be redirected to the existing `Export-API.ps1` script.

- This obviates the need for most language-specific implementations to have an embedded implementation in the DevEx platform or to write a set of wrapper scripts to transform commands to existing script calls.

Do I have that right?